themes/src/main/resources/theme/base/login/resources/js/webauthnRegister.js

```diff
@@ -0,0 +1,130 @@
+import { base64url } from "../../../common/keycloak/node_modules/rfc4648/lib/rfc4648.js";
+
+window.registerByWebAuthn = (challenge, userid, username, signatureAlgorithms, rpEntityName, rpId, attestationConveyancePreference, authenticatorAttachment, requireResidentKey, userVerificationRequirement, createTimeout, excludeCredentialIds, errmsg) => {
+    // Check if WebAuthn is supported by this browser
+    if (!window.PublicKeyCredential) {
+        document.getElementById("error").value = errmsg;
+        document.getElementById("webauth").submit();
+        return;
+    }
+
+    // mandatory parameters
+    let publicKey = {
+        challenge: base64url.parse(challenge, {loose: true}),
+        rp: {name: rpEntityName},
+        user: {
+            id: base64url.parse(userid, {loose: true}),
+            name: username,
+            displayName: username
+        },
+        pubKeyCredParams: getPubKeyCredParams(signatureAlgorithms),
+    };
+
+    // optional parameters
+    publicKey.rp.id = rpId;
```

Comment:
Why not assign this with the rest before it? Eg.

```js
rp: {id: rpId, name: rpEntityName},
```

Reply:
I fixed the codes as you suggested.